js/misc/paygManager.js

```diff
@@ -270,7 +270,7 @@ var PaygManager = new Lang.Class({
 
     _timeRemainingSecs: function() {
         if (!this._enabled)
-            return GLib.MAXUINT64;
```

Comment:
Just noticing this and leaving a drive-by comment; you may run into trouble with that, as JS doesn't have 64-bit integers, so that value will be stored as a double. You might consider using `Number.MAX_SAFE_INTEGER` as the marker instead, which is IIRC 2^53 - 1.

```
gjs> const {GLib} = imports.gi;
gjs> var b = GLib.MAXUINT64

(gjs:23920): Gjs-WARNING **: Value 18446744073709551615 cannot be safely stored in a JS Number and may be rounded
gjs> b - 1 === b
true
```

Reply:
Will, that sounds like, if we want that to be usable from JS, we need the D-Bus service to cap that property at the value of `Number.MAX_SAFE_INTEGER` as well. And, for practical reasons, we'll probably need to alias that value which isn't ideal but I guess it's reasonably well-defined for JS and shouldn't change at any point.

But this seems like a broader problem that JS introduces when binding practically any other language (but especially C). I wonder if this has been hit anywhere else in the Shell. When I noticed this problem, I didn't see any other instances of `GLib.MAXUINT64` but it could always arrive via other D-Bus services, etc.

Reply:
I can think of at least one case off the top of my head where it could break, though unlikely to ever be reached; if you have 2^53 signal connections then `obj.connect('something', func)` will start returning duplicate signal handler IDs at some point.

Reply:
In 2019's SpiderMonkey, we'll have bigints, by the way, though I'm not yet sure how to change the bindings in a backwards-compatible way.

Reply:
I think we'll have bigger problems before we quite reach 2^53 signal connections :)

Reply:
I sort of object to changing the daemon to work around this JS limitation, but I suppose it's the pragmatic approach. (I wondered briefly whether we could always store these timestamps as GVariants, but while `g_variant_compare()` does exist, there's no way to do arithmetic on them. Another way would be to move this logic to C, of course. I don't necessarily wish that on anyone.)

(2⁵³ - 1) µs is 285 years; even counting from 1970, we'll all have had our consciousnesses uploaded to the cloud long before we'd have to worry about it not been "forever" enough.